### PR TITLE
Portal extension

### PR DESCRIPTION
**Pull request by Fueredoriku**
New portal extension by @SondreHus !

### CODE REVIEW

File: Assets/Materials/Gunparts/Portal/New Material.mat
Comment (toberge) [inline Assets/Materials/Gunparts/Portal/New Material.mat:1]: Rename this file please? 🙏

File: Assets/Models/GunParts/icosphere1subdiv 1.fbx
Comment (toberge) [inline Assets/Models/GunParts/icosphere1subdiv 1.fbx:1]: Rename this to fit naming conventions and at least not have a space?

File: Assets/Prefabs/GunParts/Portal/PortalExtensionOld.prefab
Comment (toberge) [inline Assets/Prefabs/GunParts/Portal/PortalExtensionOld.prefab:1]: Remove this, since it is apparently "Old"?

File: Assets/Prefabs/Input/Player.prefab [hunk 1]
```diff
@@ -67,7 +67,7 @@ MonoBehaviour:
     m_Bits: 83
   lookSpeed: 3
   LookSpeedZoom: 0.75
-  mouseZoomSpeedFactor: 0.1
+  mouseZoomSpeedFactor: 1
```
Comment (toberge) [inline Assets/Prefabs/Input/Player.prefab:70]: Why was this changed by one order of magnitude? (doesn't feel like it's different though 🤔 )

File: Assets/Scripts/Extensions/PortalExtension/IcosphereEffectController.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    private MeshRenderer meshRenderer;
+    private Material material;
+
+    private void Start()
+    {
+        meshRenderer.materials[0] = Instantiate(meshRenderer.materials[0]);
+        material = meshRenderer.materials[0];
+    }
+
+    void Update()
```
Comment (toberge) [inline Assets/Scripts/Extensions/PortalExtension/IcosphereEffectController.cs]: private :))))))

File: Assets/Scripts/Extensions/PortalExtension/PortalExtensionController.cs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            playerView = inputManager.transform;
+        }
+    }
+
+
+    private void OnDestroy()
+    {
+        if(controller)
+            controller.AimCorrectionEnabled = true;
+        Destroy(portalMover.gameObject);
```
Comment (toberge) [inline Assets/Scripts/Extensions/PortalExtension/PortalExtensionController.cs]: `portalMover` may be destroyed already, so this needs an if guard.

File: Assets/Scripts/Extensions/PortalExtension/PortalExtensionOutput.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PortalExtensionOutput : MonoBehaviour
+{
+    [SerializeField]
+    private Transform output;
+
+    public Transform Output { get { return output; } }
```
Comment (toberge) [inline Assets/Scripts/Extensions/PortalExtension/PortalExtensionOutput.cs]: Should be `=> output;`

File: Assets/Shaders/Hoel.mat
Comment (toberge) [inline Assets/Shaders/Hoel.mat:1]: Is this a hole or Jørn Hoel? :P

File: Assets/Shaders/Portal/New Material 1.mat
Comment (toberge) [inline Assets/Shaders/Portal/New Material 1.mat:1]: Another "New Material" smh

File: Assets/Shaders/Portal/New Material 2.mat
Comment (toberge) [inline Assets/Shaders/Portal/New Material 2.mat:1]: Yet another New Material >_<

File: Assets/Shaders/Portal/New Material 3.mat
Comment (toberge) [inline Assets/Shaders/Portal/New Material 3.mat:1]: aaaaaaaaaaa

File: Assets/Shaders/Portal/New Material.mat
Comment (toberge) [inline Assets/Shaders/Portal/New Material.mat:1]: 💀

**Review by toberge — APPROVED** [review]
Looks great and works really well! I just have a bunch of nitpicks (and the portal might be a tad too small to be spotted clearly, perhaps?)

File: Assets/Prefabs/Input/Player.prefab (hunk 1 above)
Comment (Fueredoriku) [reply Assets/Prefabs/Input/Player.prefab:70]: Its broken in dev after #635 , I'm guessing because of the prefab variant rework

File: Assets/Scripts/Extensions/PortalExtension/PortalExtensionController.cs (hunk 3 above)
Comment (Fueredoriku) [reply Assets/Scripts/Extensions/PortalExtension/PortalExtensionController.cs]: Done 🤠

File: Assets/Scripts/Extensions/PortalExtension/PortalExtensionOutput.cs (hunk 4 above)
Comment (Fueredoriku) [reply Assets/Scripts/Extensions/PortalExtension/PortalExtensionOutput.cs]: Done 🤠
